packages/core/src/routes/session.test.ts

```diff
@@ -42,21 +41,23 @@ jest.mock('@/lib/passcode', () => ({
   },
 }));
 
-const MockedProvider = Provider as jest.MockedClass<typeof Provider>;
-
-const getProvider = (): Provider => {
-  const provider = MockedProvider.mock.instances[0];
-
-  if (!provider) {
-    throw new Error('Provider is not initialized');
-  }
-
-  return provider;
-};
+const grantSave = jest.fn(async () => 'finalGrantId');
+const interactionResult = jest.fn(async () => 'redirectTo');
+const interactionDetails: jest.MockedFunction<() => Promise<unknown>> = jest.fn(async () => ({}));
+jest.mock('oidc-provider', () => ({
+  Provider: jest.fn(() => ({
+    Grant: jest.fn(() => ({
```

Comment:
nit: if not spyOn Provider and Grant themselves, feels like no need to set jest.fn on the upper scope?